pkg/types/external/platform.go

```diff
@@ -0,0 +1,12 @@
+package external
+
+// Platform stores configuration related to external cloud providers.
+type Platform struct {
+	// PlatformName holds the arbitrary string representing the infrastructure provider name, expected to be set at the installation time.
+	// This field is solely for informational and reporting purposes and is not expected to be used for decision-making.
+	// +kubebuilder:default:="Unknown"
+	// +default="Unknown"
```

Comment:
I don't see this value of "Unknown" being set.

If it's not being set (i.e. I am mistaken), you could set it in the `SetPlatformDefaults` func.